src/feat/mel-computations.cc

```diff
@@ -303,5 +309,33 @@ void Lpc2Cepstrum(int n, const BaseFloat *pLPC, BaseFloat *pCepst) {
   }
 }
 
+void GetEqualLoudnessVector(const MelBanks &mel_banks,
+                            Vector<BaseFloat> *ans) {
+  int32 n = mel_banks.NumBins();
+  // central freq of each mel bin
+  const Vector<BaseFloat> &f0 = mel_banks.GetCenterFreqs();
+  ans->Resize(n);
+  for (int32 i = 0; i < n; i++) {
+    BaseFloat fsq = f0(i) * f0(i);
+    BaseFloat fsub = fsq / (fsq + 1.6e5);
+    (*ans)(i) = fsub * fsub * ((fsq + 1.44e6) / (fsq + 9.61e6));
+  }
+}
+
+
+// Compute LP coefficients from autocorrelation coefficients.
+BaseFloat ComputeLpc(const VectorBase<BaseFloat> &autocorr_in,
+                     Vector<BaseFloat> *lpc_out) {
+  int32 n = autocorr_in.Dim() - 1;
+  KALDI_ASSERT(lpc_out->Dim() == n);
+  Vector<BaseFloat> tmp(n);
+  BaseFloat ans =  Durbin(n, autocorr_in.Data(),
+                          lpc_out->Data(),
+                          tmp.Data());
+  if (ans <= 0.0)
+    KALDI_WARN << "Zero energy in LPC computation";
+  return -Log((double)1.0/ans);  // forms the C0 value
```

Comment:
I always though 1.0 was double (e. g., http://stackoverflow.com/a/4353788/1149924). Is this to support non-conformant compilers that have issues with it?